src/selectors/per-thread/composed.js

```diff
@@ -66,7 +77,24 @@ export function getComposedSelectorsPerThread(
     }
   );
 
+  /**
+   * This selector combines the marker timing and stack timing for the stack chart.
+   * This way it displays UserTiming along with the stack chart.
+   */
+  const getCombinedTimingRows: Selector<CombinedTimingRows> = createSelector(
```

Comment:
This function will need a test. I would suggest to at least check that it returns timing information with both stack information, and marker information, and that would be good enough. The other functions are pretty well tested already.

The stack chart tests appear to be a bit older, since it was an older component, and are located in: `src/test/store/actions.test.js`. I would suggest building up a profile that combines information from both`getProfileFromTextSamples` function and `getProfileWithMarkers`. The flame graph has newer tests and uses `getProfileFromTextSamples`, which is a good example to see how it's used. Plus the function should have some docs on its usage.

The marker chart timing tests are here: `src/test/store/markers.test.js`. These use `getProfileWithMarkers`.

The trick here is to:
 * Generate a stack-based profile with `getProfileFromTextSamples`
 * Generate a (user timing) marker-based profile with `getProfileWithMarkers`.
 * Set the RawMarkerTable from marker-based profile to the stack-based one.

e.g. something like.

```
stackProfile.threads[0].markers = markerProfile.threads[0].markers;
```